### PR TITLE
FlutterFragment predictive back

### PR DESCRIPTION
**Pull request by justinmc**
Following up on https://github.com/flutter/engine/pull/39208 and https://github.com/flutter/engine/pull/42789, which added predictive back support for FlutterActivity.

My test app for this is: https://github.com/justinmc/flutter-add-to-app

I tried this both with the Flutter app as the main activity, and with it as a secondary activity after an initial Android route. Predictive back worked in both cases.

Related: https://github.com/flutter/flutter/issues/109558, and my comment https://github.com/flutter/flutter/issues/109558#issuecomment-1683099477.

b/242216228

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java [hunk 1]
```diff
@@ -521,6 +521,7 @@ protected FlutterFragment createFlutterFragment() {
             ? TransparencyMode.opaque
             : TransparencyMode.transparent;
     final boolean shouldDelayFirstAndroidViewDraw = renderMode == RenderMode.surface;
+    final boolean shouldAutomaticallyHandleOnBackPressed = Build.VERSION.SDK_INT >= 33;
```
Comment (math1man) [inline shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java:524]: With this change, you'll need to update the implementation of `onBackPressed()` below. On Android 33, the `onBackPressed()` method is still relevant, but it won't behave correctly without a call to `super.onBackPressed()`. 

Alternatively, you could make this condition always true and fully remove the `onBackPressed()` override below. That way you have less conditional behavior.

Comment (math1man) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java:524]: I suppose removing the `onBackPressed()` override could impact behavior for any clients that subclass `FlutterFragmentActivity` and themselves override the method, but given that Android now formally discourages that anyways, that seems like a reasonable breaking change.

**Comment by goderbauer** [comment]
I spoke to @justinmc last week and he told me that he will get back to this one soon.

**Comment by justinmc** [comment]
I'll re-evaluate this after https://github.com/flutter/engine/pull/49093 is merged.

**Comment by justinmc** [comment]
Spiritual successor: https://github.com/flutter/engine/pull/52302

**Pull request closed without merging**